core/bin/zksync_tee_prover/src/tee_prover.rs

```diff
@@ -0,0 +1,142 @@
+use std::time::Duration;
+
+use reqwest::Client;
+use secp256k1::{ecdsa::Signature, Message, Secp256k1, SecretKey};
+use url::Url;
+use zksync_basic_types::H256;
+use zksync_node_framework::{
+    service::{ServiceContext, StopReceiver},
+    task::{Task, TaskId},
+    wiring_layer::{WiringError, WiringLayer},
+};
+use zksync_prover_interface::inputs::TeeVerifierInput;
+use zksync_tee_verifier::Verifiable;
+use zksync_types::{tee_types::TeeType, L1BatchNumber};
+
+use crate::api_client::ApiClient;
+
+/// Wiring layer for [`TeeProver`]
+///
+/// ## Requests resources
+///
+/// no resources requested
+///
+/// ## Adds tasks
+///
+/// - [`TeeProver`]
```

Comment:
Nit (here and above): This structure is private, so probably shouldn't be formatted as link.